### PR TITLE
Disabling debug support for node4, 6.10

### PR DESCRIPTION
**Pull request by awschristou**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Infrastructure (change which improves the lifecycle management (CI/CD, build, package, deploy, lint, etc) of the application, but does not change functionality.)
- [ ] Technical debt (change which improves the maintainability of the codebase, but does not change functionality)
- [ ] Testing (change which modifies or adds test coverage, but does not change functionality)
- [ ] Documentation (change which modifies or adds documentation, but does not change functionality)

## Description

Users see an error message when clicking the Debug codeLens if their lambda handler is determined to be Node 4.3 or Node 6.10
The Debug CodeLens would have been disabled ideally, but this would require loading data from the template file, further delaying the CodeLenses from appearing.

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/shared/codelens/typescriptCodeLensProvider.ts [hunk 1]
```diff
@@ -30,7 +30,12 @@ import {
     getRuntimeForLambda,
     LambdaLocalInvokeParams,
     LocalLambdaRunner,
- } from './localLambdaRunner'
+} from './localLambdaRunner'
+
+const unsupportedNodeJsRuntimes: Set<string> = new Set<string>([
+    'nodejs4.3',
+    'nodejs6.10',
```
Comment (bryceitoc9) [inline src/shared/codelens/typescriptCodeLensProvider.ts]: Double-check that `nodejs` equals `nodejs8.10` and not `nodejs0.10` (see https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html )?

File: src/shared/codelens/typescriptCodeLensProvider.ts [hunk 2]
```diff
@@ -107,10 +112,21 @@ export function initialize({
...
-            await invokeLambda({
-                runtime,
-                ...params,
-            })
+
+            if (params.isDebug && unsupportedNodeJsRuntimes.has(runtime)) {
+                vscode.window.showErrorMessage(
+                    localize(
+                        'AWS.lambda.debug.runtime.unsupported',
+                        'Debug support for {0} is currently not supported',
```
Comment (bryceitoc9) [inline src/shared/codelens/typescriptCodeLensProvider.ts]: On the fence on whether we should mention that `run` works...I can't really think of a message that doesn't sound clunky though.

Comment (awschristou) [reply src/shared/codelens/typescriptCodeLensProvider.ts]: We're pretty clear about run vs debug elsewhere in the UX.

File: src/shared/codelens/typescriptCodeLensProvider.ts (hunk 1 above)
Comment (awschristou) [reply src/shared/codelens/typescriptCodeLensProvider.ts]: I think sam cli uses nodejs8.10 as the default. 
When you run `sam init --runtime nodejs` it produces a nodejs8.10 project.
It takes the first config in the entry, [at this time](https://github.com/awslabs/aws-sam-cli/blob/develop/samcli/local/common/runtime_template.py#L33) it is nodejs8.10

**Pull request merged**